sql/core/src/test/scala/org/apache/spark/sql/connector/DatasourceV2SQLBase.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector
+
+import org.apache.spark.sql.{DataFrame, SaveMode}
+import org.apache.spark.sql.connector.catalog.{CatalogPlugin, CatalogV2Implicits, Identifier, Table}
+import org.apache.spark.sql.internal.SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION
+import org.apache.spark.util.Utils
+
+class DatasourceV2SQLBase
+  extends InsertIntoTests(supportsDynamicOverwrite = true, includeSQLOnlyTests = true)
+  with AlterTableTests {
```

Comment:
`InsertIntoTests` and `AlterTableTests` doesn't make sense for `AlterTablePartitionV2SQLSuite`. `DatasourceV2SQLBase` should only do
```
... extends QueryTest
  with SharedSparkSession
  with BeforeAndAfter
```